diff -rupN --no-dereference gcc-12.2.1-20220819/config/intdiv0.m4 gcc-12.2.1-20220819-new/config/intdiv0.m4
--- gcc-12.2.1-20220819/config/intdiv0.m4	2022-08-19 19:14:13.000000000 +0200
+++ gcc-12.2.1-20220819-new/config/intdiv0.m4	2022-08-30 09:35:05.239027520 +0200
@@ -31,10 +31,10 @@ sigfpe_handler (sig) int sig;
   exit (sig != SIGFPE);
 }
 
-int x = 1;
-int y = 0;
-int z;
-int nan;
+volatile int x = 1;
+volatile int y = 0;
+volatile int z;
+volatile int nan;
 
 int main ()
 {
diff -rupN --no-dereference gcc-12.2.1-20220819/libiberty/acinclude.m4 gcc-12.2.1-20220819-new/libiberty/acinclude.m4
--- gcc-12.2.1-20220819/libiberty/acinclude.m4	2022-08-19 19:14:13.000000000 +0200
+++ gcc-12.2.1-20220819-new/libiberty/acinclude.m4	2022-08-30 09:35:05.240027520 +0200
@@ -157,7 +157,7 @@ if test $ac_cv_os_cray = yes; then
 fi
 
 AC_CACHE_CHECK(stack direction for C alloca, ac_cv_c_stack_direction,
-[AC_TRY_RUN([find_stack_direction ()
+[AC_TRY_RUN([__attribute__ ((noclone,noinline)) find_stack_direction ()
 {
   static char *addr = 0;
   auto char dummy;
